chanjo/cli/load.py

```diff
@@ -68,14 +68,14 @@ def link(context, bed_stream):
     """Link related genomic elements."""
     chanjo_db = ChanjoDB(uri=context.obj['database'])
     result = link_elements(bed_stream)
-    with click.progressbar(result.models, length=result.count,
-                           label='adding transcripts') as bar:
-        for tx_model in bar:
-            chanjo_db.add(tx_model)
     try:
-        chanjo_db.save()
+        with chanjo_db.begin() as session:
+            with click.progressbar(result.models, length=result.count,
+                                   label='adding transcripts') as bar:
+                for tx_model in bar:
+                    session.add(tx_model)
+
     except IntegrityError:
-        LOG.exception('elements already linked?')
-        chanjo_db.session.rollback()
```

Comment:
Rollback is handled automatically by the session, see [here](https://github.com/dgilland/sqlservice/blob/113e0712a31dc40ff888737654f911776514e74f/src/sqlservice/database.py#L192)